pf/tfbridge/provider_invoke.go

```diff
@@ -96,6 +97,12 @@ func (p *provider) readDataSource(ctx context.Context, handle datasourceHandle,
 			handle.terraformDataSourceName, err)
 	}
 
+	// TODO[pulumi/pulumi#12710] consuming programs (at lest in Go and YAML) are unable to accept secrets from an
+	// Invoke response at the moment. Replace secrets with underlying plain values.
+	for k, v := range propertyMap {
+		propertyMap[k] = propertyvalue.RemoveSecrets(v)
```

Comment:
This might be too heavy handed, but it feels like we should warn here. Users will expect that secrets can be returned through invokes.
```suggestion
		if propertyValue.ContainsSecrets() {
			warn(invokeName, "Secrets in return value '%s' were dropped")
		}
		propertyMap[k] = propertyvalue.RemoveSecrets(v)
```

CC @AaronFriel

Reply:
Yes, we probably should warn.